bundle/src/main/java/com/adobe/acs/commons/fetchfile/impl/FileFetchImpl.java

```diff
@@ -0,0 +1,229 @@
+    log.info("Activing with configuration: {}", config);
+    this.config = config;
+    run();
+  }
+
+  private byte[] fetchFile() throws IOException {
+    log.trace("fetchFile");
+
+    HttpURLConnection con = null;
+    try {
+      log.debug("Opening connection to {}", config.remoteUrl());
+      con = openConnection();
+
+      List<Pair<String, String>> header = Arrays.stream(config.headers()).map(h -> h.split("\\="))
+          .filter(h -> (h.length >= 2)).map(h -> {
+            String value = Arrays.stream(ArrayUtils.remove(h, 0)).collect(Collectors.joining("="));
+            return new ImmutablePair<String, String>(h[0], value);
+          }).collect(Collectors.toList());
+      for (Pair<String, String> p : header) {
+        log.trace("Adding request property {}={}", p.getKey(), p.getValue());
+        con.addRequestProperty(p.getKey(), p.getValue());
+      }
+      con.setUseCaches(false);
+      int rc = con.getResponseCode();
+      log.debug("Retrieved response code: {}", rc);
+      if (ArrayUtils.contains(config.validResponseCodes(), rc)) {
+        log.debug("Retrieved valid response code {}", rc);
+        try (InputStream is = con.getInputStream()) {
+          return IOUtils.toByteArray(is);
+        }
+      } else {
+        log.warn("Retrieved invalid response code {}", rc);
+        throw new IOException("Failed to download file: " + rc);
+      }
+    } finally {
+      if (con != null) {
+        con.disconnect();
+      }
+    }
+  }
+
+  @Override
+  public Exception getLastException() {
+    return this.lastException;
+  }
+
+  private String hash(byte[] data) {
+    return new String(Base64.encode(DigestUtils.getMd5Digest().digest(data)));
+  }
+
+  @Override
+  public boolean isLastJobSucceeded() {
+    return this.lastJobSucceeded;
+  }
+
+  protected HttpURLConnection openConnection() throws IOException {
+    return (HttpURLConnection) new URL(config.remoteUrl()).openConnection();
+  }
+
+  @Override
+  public void run() {
+    try {
+      updateFile();
+      lastJobSucceeded = true;
+      lastException = null;
+    } catch (Exception e) {
+      log.warn("Failed to run fetch file job for {}", config.remoteUrl(), e);
+      lastException = e;
+      lastJobSucceeded = false;
+    }
+  }
+
+  public void setFactory(ResourceResolverFactory factory) {
+    this.factory = factory;
+  }
+
+  public void setReplicator(Replicator replicator) {
+    this.replicator = replicator;
+  }
+
+  @Override
+  public void updateFile() throws IOException, ReplicationException {
+    log.trace("updateFile");
+
+    try (ResourceResolver resolver = factory
+        .getServiceResourceResolver(Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, SERVICE_USER_NAME))) {
+      AssetManager manager = Optional.ofNullable(resolver.adaptTo(AssetManager.class))
+          .orElseThrow(() -> new PersistenceException("Failed to get Asset Manager"));
+
+      byte[] bin = fetchFile();
```

Comment:
Can get problematic, if the fetcher is misused for large binaries. Is it possible to use the InputStream from the HTTP response and stream the output into the repo?

Reply:
Fixed by no longer having to support generating the hash